demisto_sdk/__main__.py

```diff
@@ -423,25 +427,58 @@ def prepare_content(ctx, **kwargs):
     """
     This command is used to prepare the content to be used in the platform.
     """
-    if click.get_current_context().info_name == "unify":
-        kwargs["unify_only"] = True
-
-    check_configuration_file("unify", kwargs)
-    # Input is of type Path.
-    kwargs["input"] = str(kwargs["input"])
-    file_type = find_type(kwargs["input"])
-    if marketplace := kwargs.get("marketplace"):
-        os.environ[ENV_DEMISTO_SDK_MARKETPLACE] = marketplace.lower()
-    if file_type == FileType.GENERIC_MODULE:
-        from demisto_sdk.commands.prepare_content.generic_module_unifier import (
-            GenericModuleUnifier,
+    assert any(
+        [kwargs["all"], kwargs["input"]]
+    ), "Either '-a' or '-i' parameters must be provided."
+    assert not all(
+        [kwargs["all"], kwargs["input"]]
+    ), "Only one of '-a' and '-i' parameters should be provided."
+
+    if kwargs["all"]:
+        content_DTO = from_path()
+        content_DTO.dump(
+            dir=Path(kwargs["output"], "tmp"),
+            marketplace=parse_marketplace_kwargs(kwargs),
         )
+        return 0
 
-        # pass arguments to GenericModule unifier and call the command
-        generic_module_unifier = GenericModuleUnifier(**kwargs)
-        generic_module_unifier.merge_generic_module_with_its_dashboards()
-    else:
-        PrepareUploadManager.prepare_for_upload(**kwargs)
+    inputs = []
+    if input_ := kwargs["input"]:
+        inputs = input_.split(",")
+
+    if output_path := kwargs["output"]:
+        if "." in Path(output_path).name:  # check if the output path is a file
+            if len(inputs) > 1:
+                raise Exception(
+                    "When passing multiple inputs, the output path should be a directory and not a file."
+                )
+        else:
+            os.makedirs(output_path, exist_ok=True)
```

Comment:
create a path object for the destination path, then use `dest_path.parent.makedirs(exist_ok=True)`